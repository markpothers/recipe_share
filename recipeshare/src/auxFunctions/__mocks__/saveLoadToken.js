export function saveToken () {
	console.log("mock save token!!!!!")
	return null
}

export function loadToken () {
	return null
}
