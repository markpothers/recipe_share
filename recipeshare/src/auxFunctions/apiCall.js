import NetInfo from '@react-native-community/netinfo'

export const apiCall = async (callback, ...args) => {
    let netInfoState = await NetInfo.fetch()
    let response = {}
    if (netInfoState.isConnected) {
        try {
            response = await callback(...args)
			// console.log(response)
        } catch (e) {
            response.fail = true
        }
    } else {
        response.fail = true
    }
    return response
}
