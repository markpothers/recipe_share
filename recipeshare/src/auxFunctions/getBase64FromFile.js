import * as FileSystem from 'expo-file-system';
import * as ImageManipulator from "expo-image-manipulator";

export const getBase64FromFile = async (uri) => {
	if (uri) {
		try {
			const compressedImage = await ImageManipulator.manipulateAsync(
				uri,
				[],
				{ compress: 0.75, format: ImageManipulator.SaveFormat.JPEG }
			);
			const compressedImageBase64 = await FileSystem.readAsStringAsync(compressedImage.uri, { encoding: FileSystem.EncodingType.Base64 })
			FileSystem.deleteAsync(compressedImage.uri, { idempotent: true })
			return compressedImageBase64
		} catch {
			return ""
		}
	} else {
		return ""
	}
}
