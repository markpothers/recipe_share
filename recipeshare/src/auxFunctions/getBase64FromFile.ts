import * as ImageManipulator from "expo-image-manipulator";

export const getBase64FromFile = async (uri: string): Promise<string> => {
	if (uri) {
		try {
			const compressedImage = await ImageManipulator.manipulateAsync(uri, [], {
				base64: true,
				compress: 0.75,
				format: ImageManipulator.SaveFormat.JPEG,
			});
			// const compressedImageBase64 = await FileSystem.readAsStringAsync(compressedImage.uri, {
				// encoding: FileSystem.EncodingType.Base64,
			// });
			// console.log("compressedImageBase64:", compressedImageBase64);
			// FileSystem.deleteAsync(compressedImage.uri, { idempotent: true });
			return compressedImage.base64;
		} catch {
			return "";
		}
	} else {
		return "";
	}
};
