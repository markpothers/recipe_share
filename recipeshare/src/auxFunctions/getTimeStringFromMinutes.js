

export const getTimeStringFromMinutes = (totalMinutes) => {
	if (totalMinutes) {
		let hr = (Math.floor(totalMinutes / 60)).toString().padStart(2, '0')
		let min = (totalMinutes - (hr * 60)).toString().padStart(2, '0')
		if (hr == "NaN" || min == "NaN") {
			return "00:00"
		} else {
			return `${hr}:${min}`
		}
	} else {
		return "00:00"
	}
}

export const getMinutesFromTimeString = (timeString) => {
	if (timeString) {
		let split = timeString.split(':')
		let totalMinutes = (parseInt(split[0]) * 60) + (parseInt(split[1]))
		if (isNaN(totalMinutes)) {
			return 0
		} else {
			return totalMinutes
		}
	} else {
		return 0
	}
}