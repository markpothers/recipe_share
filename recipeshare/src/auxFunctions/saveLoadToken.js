import * as SecureStore from 'expo-secure-store';

export const saveToken = (token) => {
	let options = { keychainAccessible: SecureStore.WHEN_UNLOCKED_THIS_DEVICE_ONLY}
	SecureStore.setItemAsync('token', token, options)
}

export const loadToken = async () => {
	let token = await SecureStore.getItemAsync('token')
	if (token) {
		return token
	} else {
		return null
	}
}

export const deleteToken = () => {
	SecureStore.deleteItemAsync('token')
}
