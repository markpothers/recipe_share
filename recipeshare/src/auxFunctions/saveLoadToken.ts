import * as SecureStore from "expo-secure-store"

export const saveToken = (token: string): void => {
	const options = { keychainAccessible: SecureStore.WHEN_UNLOCKED_THIS_DEVICE_ONLY }
	SecureStore.setItemAsync("token", token, options)
}

export const loadToken = async (): Promise<string> => {
	const token = await SecureStore.getItemAsync("token")
	if (token) {
		return token
	} else {
		return null
	}
}

export const deleteToken = (): void => {
	SecureStore.deleteItemAsync("token")
}
