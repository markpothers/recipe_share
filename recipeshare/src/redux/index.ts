export { useAppDispatch, useAppSelector } from "./hooks";
export * from "./types";
export type { RootState } from "./store"
export {
	updateSingleRecipeList,
	updateAllRecipeLists,
	updateSingleChefList,
	updateAllChefLists,
	storeRecipeDetails,
	updateNewUserDetails,
	clearNewUserDetails,
	updateLoginUserDetails,
	clearLoginUserDetails,
	addRecipeLike,
	removeRecipeLike,
	addRecipeReShare,
	removeRecipeReShare,
	addRecipeMake,
	addMakePic,
	addMakePicChef,
	saveRemainingMakePics,
	updateComments,
	storeNewFollowers,
	updateLoggedInChef,
	storeChefDetails,
	clearChefDetails,
	stayLoggedIn
} from "./rootReducer"
export { default as rootReducer } from "./rootReducer"
