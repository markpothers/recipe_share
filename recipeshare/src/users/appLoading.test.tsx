import React from "react";
import { render, waitFor } from "@testing-library/react-native";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import { rootReducer } from "../redux";
import AsyncStorage from "@react-native-async-storage/async-storage";
import AppLoading from "./appLoading";
import { loadToken } from "../auxFunctions/saveLoadToken";

// manual mocks
jest.mock("../auxFunctions/saveLoadToken.ts");

describe("appLoading page", () => {
	let store, mockSetLoadedAndLoggedIn, navigation, route;

	beforeEach(async () => {
		store = configureStore({
			reducer: {
				root: rootReducer,
			},
		});

		mockSetLoadedAndLoggedIn = jest.fn();
		navigation = {};
		route = {}
	});

	afterEach(async () => {
		(AsyncStorage.getItem as jest.MockedFunction<typeof AsyncStorage.getItem>).mockClear();
		AsyncStorage.clear(); //clear out the contents of AsyncStorage between tests
	});

	test("should load and render with no saved login", async () => {
		const { toJSON } = render(
			<Provider store={store}>
				<AppLoading navigation={navigation} route={route} setLoadedAndLoggedIn={mockSetLoadedAndLoggedIn} />
			</Provider>
		);
		await waitFor(() => expect(toJSON()).toMatchSnapshot());
		expect(mockSetLoadedAndLoggedIn).toHaveBeenCalledWith({ loaded: true, loggedIn: false });
	});

	test("should load and render with saved token but no user", async () => {
		(loadToken as jest.MockedFunction<typeof loadToken>).mockImplementation(() => Promise.resolve("myMockToken"));
		render(
			<Provider store={store}>
				<AppLoading navigation={navigation} route={route} setLoadedAndLoggedIn={mockSetLoadedAndLoggedIn} />
			</Provider>
		);
		await waitFor(() => expect(mockSetLoadedAndLoggedIn).toHaveBeenCalledWith({ loaded: true, loggedIn: false }));
	});

	test("should load and render with saved token and user", async () => {
		(loadToken as jest.MockedFunction<typeof loadToken>).mockImplementation(() => Promise.resolve("myMockToken"));
		(AsyncStorage.getItem as jest.MockedFunction<typeof AsyncStorage.getItem>).mockImplementation(() =>
			Promise.resolve(
				JSON.stringify({
					id: 22,
					e_mail: "email@test.com",
					username: "testUsername",
					image_url: "testImageUrl",
					is_admin: false,
					is_member: false,
				})
			)
		);
		render(
			<Provider store={store}>
				<AppLoading navigation={navigation} route={route} setLoadedAndLoggedIn={mockSetLoadedAndLoggedIn} />
			</Provider>
		);
		await waitFor(() => expect(mockSetLoadedAndLoggedIn).toHaveBeenCalledWith({ loaded: true, loggedIn: true }));
	});
});
