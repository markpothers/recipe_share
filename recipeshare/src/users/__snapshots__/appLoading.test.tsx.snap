// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appLoading page should load and render with no saved login 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          {},
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "top": 133.4,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          {
            "height": 266.8,
            "width": "75%",
          }
        }
        testID="yellowLogo"
      />
    </View>
  </View>
</View>
`;
